engines/bastion_katello/app/assets/javascripts/bastion_katello/tasks/task.factory.js

```diff
@@ -1,264 +1,112 @@
- /**
+/**
  * @ngdoc service
  * @name  Bastion.tasks.factory:Task
  *
  * @requires BastionResource
- * @requires $timeout
- * @requires $log
- * @requires CurrentOrganization
  *
  * @description
- *   Provides a BastionResource for task(s).
- *
- *   Also provides a polling bulk search for tasks, define in
- *   foreman-tasks engine API. This allows polling for multiple task
- *   statuses with a single request (e.g. currently running tasks of a
- *   user, recent tasks of a user, tasks for given resource, detail of
- *   one task etc). See `resource.registerSearch` and
- *   `resource.unregisterSearch` for details.
+ *   Provides a BastionResource for task or list of tasks.
  */
-
-angular.module('Bastion.tasks').factory('Task',
-    ['BastionResource', '$timeout', '$log', '$q', 'CurrentOrganization',
-    function (BastionResource, $timeout, $log, $q, CurrentOrganization) {
-        var bulkSearchRunning = false, searchIdGenerator = 0,
-            searchParamsById = {}, callbackById = {}, pollCount = 0, maxPollInterval = 10000;
-
-        var resource = BastionResource('/katello/api/v2/tasks/:id/:action',
-            {id: '@id', 'organization_id': CurrentOrganization}, {});
-
-        var foremanTasksResource = BastionResource('/foreman_tasks/api/tasks/:id/:action',
-            {},
+angular.module('Bastion.tasks').factory('Task', [
+    '$timeout',
+    'BastionResource',
+    'CurrentOrganization',
+    function($timeout, BastionResource, CurrentOrganization) {
+        var pendingTimeout;
+        var productId;
+
+        var resource = BastionResource(
+            '/foreman_tasks/api/tasks/',
+            {
+                'organization_id': CurrentOrganization,
+                isArray: true
+            },
             {
-                bulkSearch: {method: 'POST', isArray: true, params: { action: 'bulk_search'}}
+                update: { method: 'PUT' },
+                sync: { method: 'POST', params: { action: 'sync' } },
+                updateSyncPlan: {
```

Comment:
Why does this have `updateSyncPlan` and `sync` on the `Task` resource?

Reply:
these we're just a copy from another factory. They've been removed.